src/java/htsjdk/samtools/SAMRecord.java

```diff
@@ -447,24 +423,18 @@ public String getMateReferenceName() {
      * mate reference index to NO_ALIGNMENT_REFERENCE_INDEX.
      *
      * @param mateReferenceName - must not be null
+     * @throws IllegalArgumentException if {@code mateReferenceName} is null
      */
     public void setMateReferenceName(final String mateReferenceName) {
         if (null == mateReferenceName) {
-            throw new IllegalArgumentException("Mate reference name must not be null");
+            throw new IllegalArgumentException("Mate reference name must not be null. Use SAMRecord.NO_ALIGNMENT_REFERENCE_NAME to reset the mate reference name.");
         }
-        else if (NO_ALIGNMENT_REFERENCE_NAME.equals(mateReferenceName)) {
-            mMateReferenceName = NO_ALIGNMENT_REFERENCE_NAME;
-            mMateReferenceIndex = NO_ALIGNMENT_REFERENCE_INDEX;
```

Comment:
Same comment as for `setReferenceName()` above -- the behavior has changed so that if the header is null and `NO_ALIGNMENT_REFERENCE_NAME` is passed in here, `mMateReferenceIndex` ends up as null rather than `NO_ALIGNMENT_REFERENCE_INDEX`. Is this desired?